### PR TITLE
Added the users participation history to volunteerProfile

### LINKED ISSUE #664: Add Volunteer's history
On a users Volunteer Profile we need to have an area that shows all the events they have participated in.

### PR DESCRIPTION
**Pull request by AndersonStettner**

### CODE REVIEW

**Comment by AndersonStettner** [comment]
[Add Volunteer's history #664](https://trello.com/c/G8D7OQYt/463-add-volunteers-history-664)

File: app/logic/events.py [hunk 1]
```diff
@@ -171,16 +172,16 @@ def getOtherEvents(term):
...
-                    .order_by(Event.id)
-                  )
+                        .join(ProgramEvent, JOIN.LEFT_OUTER)
+                        .join(Program, JOIN.LEFT_OUTER)
+                        .where(Event.term == term,
+                               Event.isTraining == False,
+                               Event.isAllVolunteerTraining == False,
+                               ((ProgramEvent.program == None) |
+                                (Program.isStudentLed == False) &
+                                (Program.isBonnerScholars == False)))
```
Comment (BrianRamsay) [inline app/logic/events.py]: We should put parentheses around the `Program.` pair so we are explicit about the order.

File: app/controllers/events/routes.py [hunk 2]
```diff
@@ -12,15 +12,8 @@
 from app.controllers.events import events_bp
 from app.controllers.events import email
 from app.logic.emailHandler import EmailHandler
-from app.logic.events import getUpcomingEventsForUser
 from app.logic.participants import sendUserData
 
-@events_bp.route('/events/upcoming_events', methods=['GET'])
-def showUpcomingEvent():
-    upcomingEvents = getUpcomingEventsForUser(g.current_user)
-    return render_template('/events/showUpcomingEvents.html',
```
Comment (BrianRamsay) [inline app/controllers/events/routes.py:21]: You should get rid of the unused template as well

File: tests/code/test_events.py [hunk 3]
```diff
@@ -546,6 +547,60 @@ def test_upcomingEvents():
...
+        ProgramEvent.create(program = participatedProgram, event = participatedProgramEvent)
+
+        # Add the created user as a participnt to the created program event
+        EventParticipant.create(user = user , event = participatedProgramEvent.id)
+        assert participatedProgramEvent in getParticipatedEventsForUser(user)
+
+        # Add the created user as a participant to the create non-program event
+        EventParticipant.create(user = user, event = participatedEvent.id)
+        assert participatedEvent in getParticipatedEventsForUser(user)
+
```
Comment (BrianRamsay) [inline tests/code/test_events.py:601]: I would check an event that SHOULDN'T be returned as well, to make sure your query isn't just getting a bunch of extra events.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

File: app/models/event.py [hunk 4]
```diff
@@ -32,15 +32,13 @@ def noProgram(self):
...
-            try:
-                countPE = list(self.programEvents.select(ProgramEvent, Program).join(Program).execute())
-                if len(countPE) == 1:
-                    self._spCache = countPE[0].program
-                else:
-                    self._spCache = None
-            except DoesNotExist:
-                self._spCache = self
+        
+        if self._spCache is "Empty":
```
Comment (BrianRamsay) [inline app/models/event.py]: Use `==` instead of `is`

Comment (BrianRamsay) [reply app/models/event.py]: This warning shows up in the `flask run` output.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**